### PR TITLE
A few fixes and updates

### PR DESCRIPTION
**Pull request by snukky**
* Fixing #374 
* Fixing https://github.com/marian-nmt/marian/issues/217
* Fixing #338 
* Unifying some logging messages
* Logging debugging messages in `marian.cpp` at 'debug' log level

### CODE REVIEW

File: src/command/marian.cpp [hunk 1]
```diff
@@ -21,14 +21,14 @@ int main(int argc, char** argv) {
   // This means we have two redundant implementations of multi-node sync-sgd. Note that the
   // MultiNodeGraphGroup family is out of date. Therefore, the goal is to remove MultiNodeGraphGroupSync.
   if(options->get<bool>("multi-node")) {
-    LOG(warn, "[experimental] Old multi-node training implementations. These are presently not up-to-date.");
+    LOG(warn, "[experimental] Old multi-node training implementations. These are presently not up-to-date");
 
     if(options->get<bool>("sync-sgd")) {
-      LOG(warn, "[training] Using MultiNodeGraphGroupSync trainer.");
+      LOG(debug, "[training] Using MultiNodeGraphGroupSync trainer");
```
Comment (emjotde) [inline src/command/marian.cpp]: info instead of debug?

Comment (emjotde) [reply src/command/marian.cpp]: And for all below

File: src/common/config_parser.cpp [hunk 2]
```diff
@@ -560,13 +561,16 @@ void ConfigParser::addSuboptionsDevices(cli::CLIWrapper& cli) {
...
   // clang-format off
   cli.add<int>("--mini-batch",
-      "Size of mini-batch used during update",
-      defaultMiniBatch);
+               (mode_ == cli::mode::translation)
+                   ? "Size of mini-batch used during batched translation" :
+               (mode_ == cli::mode::scoring)
+                   ? "Size of mini-batch used during batched scoring"
+                   : "Size of mini-batch used during update",
+               defaultMiniBatch);
```
Comment (emjotde) [inline src/common/config_parser.cpp]: Comment please

File: src/common/config_validator.cpp [hunk 3]
```diff
@@ -119,13 +118,25 @@ void ConfigValidator::validateOptionsTraining() const {
...
 
+void ConfigValidator::validateModelExtension(cli::mode mode) const {
+  std::vector<std::string> models;
+  if(mode == cli::mode::translation)
+    models = get<std::vector<std::string>>("models");
+  else
+    models.push_back(get<std::string>("model"));
+
+  for(const auto& modelPath : models)
+    UTIL_THROW_IF2(!(utils::endsWith(modelPath, ".npz") || utils::endsWith(modelPath, ".bin")),
```
Comment (emjotde) [inline src/common/config_validator.cpp]: THROW?

File: src/common/config_validator.cpp [hunk 4]
```diff
@@ -143,11 +154,7 @@ void ConfigValidator::validateDevices(cli::mode mode) const {
   }
 
   UTIL_THROW_IF2(!regex::regex_match(devices, pattern),
-                 "the argument '(" + devices
-                     + ")' for option '--devices' is invalid. "
-                     + help);
-
-
+                 "the argument '(" + devices + ")' for option '--devices' is invalid. " + help);
```
Comment (emjotde) [inline src/common/config_validator.cpp]: THROW probably copied from here. Is there any reason these should be exceptions?

**Comment by snukky** [comment]
All comments addressed.

**Pull request merged**